src/Symfony/Component/Scheduler/Schedule.php

```diff
@@ -29,8 +30,13 @@ final class Schedule implements ScheduleProviderInterface
      */
     public function add(RecurringMessage $message, RecurringMessage ...$messages): static
     {
-        $this->messages[] = $message;
-        $this->messages = array_merge($this->messages, $messages);
+        foreach ([$message, ...$messages] as $m) {
+            if (isset($this->messages[$m->getId()])) {
+                throw new LogicException('Duplicated schedule message.');
```

Comment:
If using class names, I don't think we can have duplicates.

Reply:
I'm not sure what you mean here:
```php
$schedule = new Schedule();
$schedule->add(RecurringMessage::cron('* * * * *', new \stdClass()));
$schedule->add(RecurringMessage::cron('* * * * *', new \stdClass())); // is duplicate
```